lego/components.py

```diff
@@ -102,10 +122,20 @@ def run_command(self, command: str) -> str:
             The output of the command.
         """
 
-        return self._rpyc.modules["subprocess"].check_output(command.split())
+        return self.rpyc.modules["subprocess"].check_output(command.split())
 
     def get_ip(self) -> str:
-        """Gets the IP of the remote machine."""
-
-        hostname = self._rpyc.modules['socket'].gethostname()
-        return self._rpyc.modules['socket'].gethostbyname(hostname)
+        """Gets IP for default route interface of the remote machine."""
+
+        r_socket = self.get_remote_socket(socket.AF_INET, socket.SOCK_DGRAM)
+        try:
+            # Doesn't have to be reachable.
+            r_socket.connect(('10.255.255.255', 1))
+            component_ip = r_socket.getsockname()[0]
+        except socket.error:
+            # Can't find default route, use localhost interface.
+            component_ip = '127.0.0.1'
+        finally:
+            r_socket.close()
+
+        return component_ip
```

Comment:
Return all the available ip addresses. use `ipaddress` module

Reply:
You can't return all available ip addresses with only python's standard library. You can do that with specific windows/linux shell commands.